### PR TITLE
feat: LSDV-4664: Implementing file delete

### PR DESCRIPTION
**Pull request by wesleylima**
### PR fulfills these requirements
- [X] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [X] Tests for the changes have been added/updated (for bug fixes/features)
- [X] Docs have been added/updated (for bug fixes/features)
- [X] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [X] Self-reviewed and ran all changes on a local instance (for bug fixes/features)

#### Change has impacts in these area(s)
_(check all that apply)_
- [ ] Product design
- [ ] Backend (Database)
- [X] Backend (API)
- [ ] Frontend

### Describe the reason for change
Exports need to be deleted from the persistent storage when the snapshot is deleted

#### What does this fix?
_(if this is a bug fix)_

#### What is the new behavior?
The actual stored file is deleted when the snapshot is deleted

#### What is the current behavior?
The export snapshot file in the persistent storage remains even after the snapshot is deleted

#### What libraries were added/updated?
n/a

#### Does this change affect performance?
no

#### Does this change affect security?
_(if so describe the impacts positive or negative)_

#### What alternative approaches were there?
_(briefly list any if applicable)_

#### What feature flags were used to cover this change?
ff_back_dev_4664_remove_storage_file_on_export_delete_29032023_short

### Does this PR introduce a breaking change?
_(check only one)_
- [X] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [ ] No
- [ ] Not sure (briefly explain the situation below)

### What level of testing was included in the change?
_(check all that apply)_
- [X] e2e
- [ ] integration
- [ ] unit

### Which logical domain(s) does this change affect?
Project export

### CODE REVIEW

File: label_studio/data_export/api.py
Comment (makseq) [inline label_studio/data_export/api.py]: Let's add str(e) inside of this text: 
```
Could not delete file from storage. Check that your user has permissions to delete files: {str(e)}
```
where 

```
except Exception as e
```

File: label_studio/data_export/api.py
Comment (makseq) [inline label_studio/data_export/api.py:391]: seems it doesn't delete converted files?

**Review by makseq — CHANGES_REQUESTED** [review]

File: label_studio/data_export/api.py
Comment (wesleylima) [reply label_studio/data_export/api.py:391]: I didn't know that we were saving converted files to storage. It doesn't seem like we save any reference to the converted files so they're hard to locate in this process.

I'm thinking we postpone doing the converted files deletion until after https://github.com/heartexlabs/label-studio/pull/3899/ is merged. Sergei's async changes store references to the converted files in the database now so it'll be much easier to do deletion that way.

Thoughts?

Comment (makseq) [reply label_studio/data_export/api.py:391]: Yes, exactly because of this reason I put a blocker-ticket with Async conversion to LSDV-4664. So, I think we should wait until the merge. However, I've seen 1 of 2 backend PRs is merged already.

**Review by makseq — CHANGES_REQUESTED** [review]

File: label_studio/data_export/models.py
Comment (makseq) [inline label_studio/data_export/models.py]: please don't use signals, write it explicitly in the ConvertedFormat model

**Review by makseq — CHANGES_REQUESTED** [review]

File: label_studio/data_export/api.py
Comment (triklozoid) [reply label_studio/data_export/api.py:391]: Converted files deletion also implemented.

File: label_studio/data_export/models.py
Comment (triklozoid) [reply label_studio/data_export/models.py]: Fixed

File: label_studio/data_export/models.py
Comment (makseq) [inline label_studio/data_export/models.py]: do we need it?

**Review by makseq — APPROVED** [review]

**Review by AndrejOros — APPROVED** [review]

**Pull request merged**
